site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -477,6 +477,32 @@ public function showStatus($gradeable_id) {
             $this->core->redirect($this->core->buildCourseUrl());
         }
 
+        //get all graded gradeables for queue stats
+        $submissions_in_queue = 0;
+        $submissions_grading_in_progress = 0;
+        $order = new GradingOrder($this->core, $gradeable, $this->core->getUser(), true);
+        $order->sort("id", "ASC");
+        $graded_gradeables = [];
+        $user_ids = $this->core->getQueries()->getUsersOnTeamsForGradeable($gradeable); // Collect user ids so we know who isn't on a team
+        foreach ($order->getSortedGradedGradeables() as $g) {
+            $graded_gradeables[] = $g;
+            if ($gradeable->isTeamAssignment()) {
+                $user_ids = array_merge($user_ids, $g->getSubmitter()->getTeam()->getMemberUserIds());
```

Comment:
Why do you create and add to user ids? These never seem to be used late in the function. Also all these additions should probably be put beneath the access control check since there is no point in running a pretty big loop until we know they have permission to see the page.